tests: mem_protect: add error test case of userspace

enjiamai opened the pull request:
Add some error test cases for userspace of mem protect module, 
in order to improve better code coverage in our module testing.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

LeiW000 commented on tests/kernel/mem_protect/mem_map/src/main.c:
```diff
@@ -165,13 +165,28 @@ void test_z_phys_map_side_effect(void)
 }
 
+/**
+ * Show that mapping works failed as expected
```
Please add "@brief" at begining of the line.

enjiamai replied:
@LeiW000 Got it fix, this test case conflict due to another test case merged. I will put them in another commit.

LeiW000 commented on tests/kernel/mem_protect/userspace/src/main.c:
```diff
@@ -528,13 +528,13 @@ static void umode_enter_func(void)
-*
-* @ingroup kernel_memprotect_tests
-*/
+ * @brief Test to check supervisor thread enter one-way to usermode
```
It better to change it like below.
Check if supervisoer thread can enter usermode with one-way operation.

enjiamai replied:
@LeiW000  Sorry for reply late. Got it fix, thanks!

dcpleung commented on tests/kernel/mem_protect/userspace/src/main.c:
```diff
@@ -912,6 +913,381 @@ void test_tls_pointer(void)
+	struct z_object *ko;
+
+	ko = z_dynamic_object_create(INT_MAX);
+	zassert_is_null(ko, "success is not reasonable");
```
This message is confusing if you only look at serial output when this fails. Please state exactly what is wrong.

enjiamai replied:
Got it, indeed it is confusing. Thanks for correction this^^

enjiamai commented on tests/kernel/mem_protect/userspace/src/main.c:
```diff
@@ -912,6 +913,381 @@ void test_tls_pointer(void)
+	void *ko;
+	struct k_thread *curr_thread = k_current_get();
+
+#if !defined(CONFIG_X86_64)
```
Hi, it seems like that k_obj_alloc(K_OBJ_THREAD) still return NULL, give me more time to check it, thanks!

Commit:
tests: mem_protect: add error test case of userspace

Add some error test cases for userspace of memory protection module.
This increase the code coverage of testing.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>
